### PR TITLE
add api to remove WFs using correlation id

### PR DESCRIPTION
**Pull request by techyragu**
Signed-off-by: Rahul Gupta <ragupta4@cisco.com>

### CODE REVIEW

File: jersey/src/main/java/com/netflix/conductor/server/resources/WorkflowBulkResource.java [hunk 1]
```diff
@@ -116,4 +112,19 @@ public BulkResponse retry(List<String> workflowIds) {
...
+     * @param correlationId - correlationId of the workflows
+     * @param archiveWorkflow - flag to specify whether to archive a workflow, by default true
+     * @return bulk response object containing a list of succeeded workflows and a list of failed ones with errors
+     */
+    @DELETE
+    @Path("/correlationId/{correlationId}")
+    @ApiOperation("Remove workflows for a given correlation id")
+    @Consumes(MediaType.WILDCARD)
+    public BulkResponse removeCorrelatedWorkflows(@PathParam("correlationId") String correlationId,
+                               @QueryParam("archiveWorkflow") @DefaultValue("true") boolean archiveWorkflow) {
```
Comment (manjurad) [inline jersey/src/main/java/com/netflix/conductor/server/resources/WorkflowBulkResource.java:127]: Have you tested to make sure without passing archiveWorkflow, it does get archived in ES? 
Have you tested with the archiveWorkflow query param, explicitily set to true and false, and validation with ES data/

File: core/src/main/java/com/netflix/conductor/service/WorkflowBulkServiceImpl.java [hunk 2]
```diff
@@ -143,4 +150,43 @@ public BulkResponse terminate(List<String> workflowIds, String reason) {
...
+    public BulkResponse removeCorrelatedWorkflows(String correlationId, boolean archiveWorkflow) {
+        BulkResponse bulkResponse = new BulkResponse();
+        String query = String.format("correlationId=\"%s\"", correlationId);
+        SearchResult<String> result = executionDAOFacade.searchWorkflows(query, "*", 0, 5000, null);
+        result.getResults().stream().parallel().forEach(workflowId -> {
+            try {
+                executionDAOFacade.removeWorkflow(workflowId, archiveWorkflow);;
+                bulkResponse.appendSuccessResponse(workflowId);
+            } catch (Exception e) {
+                LOGGER.error("bulk terminate exception, workflowId {}, message: {} ", workflowId, e.getMessage(), e);
```
Comment (manjurad) [inline core/src/main/java/com/netflix/conductor/service/WorkflowBulkServiceImpl.java:185]: Does logging exception, log the entire traceback? Due to polling we could be repeatedly trying to remove a workflow and if the removal fails, we could end up with a lot of logs. Please be carefuly with conductor logging

File: core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java [hunk 3]
```diff
@@ -250,6 +250,7 @@ public void removeFromPendingWorkflow(String workflowType, String workflowId) {
      * @param archiveWorkflow if true, the workflow will be archived in the {@link IndexDAO} after removal from  {@link ExecutionDAO}
      */
     public void removeWorkflow(String workflowId, boolean archiveWorkflow) {
+        LOGGER.info("remove workflow {}", workflowId);
```
Comment (manjurad) [inline core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java:253]: log this after successful removal, we don;t want this log to show up for every API attempt to remove a workflow

File: core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java [hunk 4]
```diff
@@ -642,7 +642,7 @@ public Task getPendingTaskByWorkflow(String taskReferenceName, String workflowId
      */
     @VisibleForTesting
     void completeWorkflow(Workflow wf) {
-        LOGGER.debug("Completing workflow execution for {}", wf.getWorkflowId());
+        LOGGER.info("Completing workflow execution for {}", wf.getWorkflowId());
```
Comment (manjurad) [inline core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java:645]: you see value in this log and the log below?

**Review by manjurad — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/netflix/conductor/service/WorkflowBulkServiceImpl.java (hunk 2 above)
Comment (techyragu) [reply core/src/main/java/com/netflix/conductor/service/WorkflowBulkServiceImpl.java:185]: No, not the entire trace back.

File: core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java (hunk 3 above)
Comment (techyragu) [reply core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java:253]: Removed

File: core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java (hunk 4 above)
Comment (techyragu) [reply core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java:645]: Changed to debug

**Review by manjurad — APPROVED** [review]

**Review by manjurad — APPROVED** [review]

**Pull request merged**